gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/NewDagManagerBoilerPlate.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.gobblin.service.modules.orchestration;
+
+
+import java.util.Optional;
+
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+import org.apache.gobblin.annotation.Alpha;
+import org.apache.gobblin.runtime.api.MultiActiveLeaseArbiter;
+
+
+@Alpha
+@Slf4j
+@AllArgsConstructor
+public class NewDagManagerBoilerPlate {
+  private DagTaskStream dagTaskStream;
+  private DagProcFactory dagProcFactory;
+  private DagManager.DagManagerThread [] dagManagerThreads;
+  private MultiActiveLeaseArbiter multiActiveLeaseArbiter;
+  //TODO instantiate DMT
+
+  @AllArgsConstructor
+  public static class DagManagerThread implements Runnable {
```

Comment:
this class probably needn't be `public`, since I don't expect to find use outside the enclosing one.  also no need to recapitulate the parent's name in this one's.  I'm thinking:
```
private static class Thread implements Runnable
```
does that work?

Reply:
yes it does... I have updated the code in my latest commit!